### PR TITLE
Fix external source reader pybind call

### PR DESCRIPTION
**Pull request by swetha097**
This PR fixes the error with pybind call for external source reader

### CODE REVIEW

**Review by rrawther — APPROVED** [review]

**Comment by swetha097** [comment]
> @swetha097 in the python [external_source_reader.py](https://github.com/ROCm/rocAL/blob/develop/tests/python_api/external_source_reader.py), please add an option to the user to pass CPU/GPU. Please try the app also. It hangs for me

Added option to pass cpu/gpu and batch_size

File: tests/python_api/external_source_reader.py [hunk 1]
```diff
@@ -7,14 +7,17 @@
 import numpy as np
 import cupy as cp
```
Comment (LakshmiKumar23) [inline tests/python_api/external_source_reader.py:8]: move this to line 44 if statement. We do not want cpu to fail if cupy is not available
```
if device == "gpu":
```

Comment (LakshmiKumar23) [reply tests/python_api/external_source_reader.py:8]: @swetha097 can you please resolve this? i cannot run this sample without cupy on my system otherwise

Comment (swetha097) [reply tests/python_api/external_source_reader.py]: Done

**Comment by swetha097** [comment]
Addressed the comment @LakshmiKumar23

**Review by rrawther — DISMISSED** [review]
Is this supposed to be merged after last_batch_policy PR?

**Comment by sbavasab** [comment]
> Is this supposed to be merged after last_batch_policy PR?

No this is an independent PR.

File: tests/python_api/external_source_reader.py [hunk 2]
```diff
@@ -5,16 +5,18 @@
...
 
 def main():
-    batch_size = 3
+    if  len(sys.argv) < 3:
+        print ('Please pass cpu/gpu batch_size')
+        exit(0)
+    batch_size = int(sys.argv[2])
+    device = "cpu" if sys.argv[1] == "cpu" else "gpu"
     data_dir = os.environ["ROCAL_DATA_PATH"] + \
         "rocal_data/coco/coco_10_img/train_10images_2017/"
```
Comment (LakshmiKumar23) [inline tests/python_api/external_source_reader.py:19]: Not sure how the app is working for you @swetha097 This path does not exist. https://github.com/ROCm/MIVisionX-data/tree/main/rocal_data/coco/coco_10_img/images
Please update and retest

Comment (swetha097) [reply tests/python_api/external_source_reader.py:19]: Updated the path.

**Review by LakshmiKumar23 — APPROVED** [review]
can be merged

**Pull request merged**